web/themes/custom/novel/novel.theme

```diff
@@ -5,21 +5,25 @@
  * Novel Theme.
  */
 
+use Drupal\Core\Url;
+
 /**
  * Implements template_preprocess_page().
  *
  * @param mixed[] $variables
  *   Theme variables.
  */
 function novel_preprocess_page(array &$variables): void {
+  $search_result_url = Url::fromRoute('dpl_search.result')->toString();
   $data = [
-    // @todo Something Insert the right search result page url when it is ready.
-    'search-header-url' => '/search',
-    'alt-text' => t('Search field', [], ['context' => 'Search Header']),
-    'input-placeholder' => t('Start typing in order to search', [], ['context' => 'Search Header']),
+    'search-header' => [
+      'search-header-url' => $search_result_url,
+      'alt-text' => t('Search field', [], ['context' => 'Search Header']),
+      'input-placeholder' => t('Start typing in order to search', [], ['context' => 'Search Header']),
+    ],
   ];
 
   $variables['search'] = [
-    'header' => dpl_react_render('search-header', $data),
+    'header' => dpl_react_render('search-header', $data['search-header']),
```

Comment:
Consider if it would be preferable to move this code to the new `dpl_search` module. Perhaps it has more to do with the search than the theme.